spec/lita/adapters/irc/cinch_plugin_spec.rb

```diff
@@ -98,4 +100,49 @@
       subject.on_room_join(m)
     end
   end
+
+  describe "#on_room_part" do
+    before do
+      allow(Lita::Room).to receive(:create_or_update).and_return(room)
+      allow(Lita::User).to receive(:find_by_name).and_return(user)
+    end
+
+    it "triggers a part event when someone parts a channel" do
+      expect(robot).to receive(:trigger).with(
+        :user_parted_room,
+        user: user,
+        room: room,
+      )
+      subject.on_room_part(m)
+    end
+  end
+
+  describe "#on_quit" do
+    before do
+      allow(Lita::User).to receive(:find_by_name).and_return(user)
+    end
+
+    it "triggers a disconnect event when someone quits from the IRC server" do
+      expect(robot).to receive(:trigger).with(
+        :user_disconnected,
+        user: user,
+      )
+      subject.on_quit(m)
+    end
+  end
+
+  describe "#on_nick_change" do
+    before do
+      allow(Lita::User).to receive(:find_by_name).and_return(user)
```

Comment:
This question makes me realize that "user" and "from_user" are confusing names. What is the difference between each? I'm guessing from_user is the old nickname and user is the new one. Maybe new_user and old_user instead, then?